pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -39,26 +39,18 @@ type Endpoint struct {
 // endpoints `location` from the `ref` and creates a new named reference from it.
 // The function errors if the newly created reference is not parsable.
 func (e *Endpoint) RewriteReference(ref reference.Named, prefix string) (reference.Named, error) {
-	if ref == nil {
```

Comment:
I'm personally not a fan of removing this check since it is a public API and crashing would be no option for me. But this is just my personal non-blocking opinion.

Reply:
Moved to https://github.com/containers/image/pull/628#discussion_r283525478 .